content/courses/advanced-ada/parts/resource_management/anonymous_access_types.rst

```diff
@@ -252,21 +254,553 @@ cover more specific details on anonymous access types.
+object. The :ada:`Free` procedure deallocates this object.
+
+In the body of the :ada:`Hidden_Anonymous_Allocation` package, we use the named
+access type :ada:`T_Access` to handle the actual memory allocation and
+deallocation. As expected, because those operations happen on the pool
+associated with the :ada:`T_Access` type, we don't have to worry about
+potential deallocation issues.
+
+Finally, we can instantiate this package for the type we want to have
+anonymous access types for, say a type named :ada:`Rec`. Then, when using
+the :ada:`Rec` type in the main subprogram, we can simply call the
+corresponding subprograms for memory allocation and deallocation. For
+example:
+
+.. code:: ada compile_button project=Courses.Advanced_Ada.Resource_Management.Anonymous_Access_Types.Anonymous_Access_To_Object_Types.Hidden_Alloc_Dealloc_Anonymous_Access_To_Object
+
+    with Hidden_Anonymous_Allocation;
+
+    package Info is
+
+       type Rec is private;
+
+       function New_Rec return not null access Rec;
+
+       procedure Free (Obj : access Rec);
+
+    private
+
+       type Rec is record
+          I : Integer;
+       end record;
+
+       package Rec_Allocation is new
+         Hidden_Anonymous_Allocation (T => Rec);
+
+       function New_Rec return not null access Rec
+         renames Rec_Allocation.New_T;
+
+       procedure Free (Obj : access Rec)
+         renames Rec_Allocation.Free;
+
+    end Info;
+
+    with Info; use Info;
+
+    procedure Show_Info_Allocation_Deallocation is
+       RA : constant not null access Rec := New_Rec;
+    begin
+       Free (RA);
+    end Show_Info_Allocation_Deallocation;
+
+In this example, we instantiate the :ada:`Hidden_Anonymous_Allocation` package
+in the :ada:`Info` package, which also defines the :ada:`Rec` type. We
+associate the  :ada:`New_T` and :ada:`Free` subprograms with the :ada:`Rec`
+type by using subprogram renaming. Finally, in the
+:ada:`Show_Info_Allocation_Deallocation` procedure, we use these subprograms
+to allocate and deallocate the type.
+
+Possible solution using the stack
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Another approach that we could consider to avoid memory deallocation issues
+for anonymous access-to-object types is by simply using the stack for the
+object allocation. For example:
+
+.. code:: ada run_button project=Courses.Advanced_Ada.Resource_Management.Anonymous_Access_Types.Anonymous_Access_To_Object_Types.Deallocation_Anonymous_Access_To_Object_2
+
+    procedure Show_Automatic_Deallocation is
+       I  : aliased Integer;
+       --   ^ Allocating object on the stack
+
+       IA : access Integer;
+    begin
+       IA := I'Access;
+       --  Indirect allocation on the stack.
+
+       IA.all := 30;
+
+       --  Automatic deallocation at the end of the
+       --  procedure because the integer variable is
+       --  on the stack.
+    end Show_Automatic_Deallocation;
+
+In this case, we allocate the :ada:`I` object on the stack by simply declaring
```

Comment:
Maybe don't say that we allocate the object, since that term is dedicated to the notional heap. Maybe use "create" instead?